### PR TITLE
feat: dynamic group links + flexible schedule

### LINKED ISSUE #5989: [CAL-585] dynamic group links + flexible schedule
instead of showing the 3 event-types with different time, it should instead directly show the booking page with 3 times to pick from

now that we can allow multiple schedules, we can get rid of this screen:

instead, we can do [cal.com/peer+bailey](http://cal.com/peer+bailey) which immediately shows this:

but with 3 different durations (15, 30, 60)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-585](https://linear.app/calcom/issue/CAL-585/dynamic-group-links-flexible-schedule)</sub>

### PR DESCRIPTION
**Pull request by hussamkhatib**
## What does this PR do?

Tldr: for dynamic meetings eg [cal.com/peer+bailey](http://cal.com/peer+bailey), the first screen(selecting duration) is skipped and redirected to booking page, where they have the option to change the length of the call. 

Fixes #5989

// Update: dropdown replaced with badge, see comments below 
https://github.com/calcom/cal.com/assets/52914487/e8c118ae-f0ef-47e2-a483-7b89ffa7779f

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

## How should this be tested?
- Check the video attached 

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/pages/[user]/[type].tsx [hunk 1]
```diff
@@ -242,7 +242,7 @@ async function getDynamicGroupPageProps(context: GetServerSidePropsContext) {
   const length = parseInt(typeParam);
   const eventType = getDefaultEvent("" + length);
 
-  const users = await prisma.user.findMany({
+  const usersWithoutAvatar = await prisma.user.findMany({
```
Comment (hussamkhatib) [inline apps/web/pages/[user]/[type].tsx:245]: Check #7328, for why this was done.

File: apps/web/pages/[user]/[type].tsx [hunk 2]
```diff
@@ -279,16 +278,20 @@ async function getDynamicGroupPageProps(context: GetServerSidePropsContext) {
...
   if (!users.length) {
     return {
       notFound: true,
     };
   }
 
   // sort and be in the same order as usernameList so first user is the first user in the list
-  let sortedUsers: typeof users = [];
   if (users.length > 1) {
-    sortedUsers = users.sort((a, b) => {
```
Comment (hussamkhatib) [inline apps/web/pages/[user]/[type].tsx:291]: `sortedUsers` is redundant, as `users` is mutated anyway.
see [apps/web/pages/[user].tsx](https://github.com/calcom/cal.com/blob/main/apps/web/pages/%5Buser%5D.tsx#L308C5-L308C5)

File: apps/web/components/booking/BookingDescription.tsx [hunk 3]
```diff
@@ -102,6 +106,28 @@ const BookingDescription: FC<Props> = (props) => {
             </div>
           </div>
         )}
+        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+        {/* @ts-ignore */}
```
Comment (hussamkhatib) [inline apps/web/components/booking/BookingDescription.tsx:110]: The type definition for `eventType` is really complex, Please do help If anyone knows how to deal with it.

**Comment by CarinaWolli** [comment]
@hussamkhatib There are some type errors, could you take a look at that?

**Comment by hussamkhatib** [comment]
@CarinaWolli Looks like the Booking component is replaced with a new one, I will need to re-do it. I'll make sure to fix the type Error as well. Turning this into a draft.

**Comment by PeerRich** [comment]
we already have the logic and UI to have multiple durations, can you use that instead of the dropdown?
image

**Comment by hussamkhatib** [comment]
- Replaced dropdown with the new UI.
- Marked previous comments as resolved as previous commits went to trash.

File: packages/features/bookings/components/event-meta/Duration.tsx [hunk 4]
```diff
@@ -1,37 +1,65 @@
...
     <div className="flex flex-wrap gap-2">
-      {event.metadata.multipleDuration.map((duration) => (
-        <Badge
-          variant="gray"
-          className={classNames(selectedDuration === duration && "bg-brand-default text-brand")}
-          size="md"
-          key={duration}
-          onClick={() => setSelectedDuration(duration)}>{`${duration} ${t("minute_timeUnit")}`}</Badge>
-      ))}
+      {event.metadata?.multipleDuration &&
```
Comment (hussamkhatib) [inline packages/features/bookings/components/event-meta/Duration.tsx:34]: `Boolean(event.metaData?.,multipleDuration)` and `isDynamicEvent` can never be true at the same time, so we won't render the same badge again from line 43.
// see `packages/trpc/server/routers/publicViewer/event.handler.ts`

**Comment by emrysal** [comment]
I've made a few changes, such as deprecating 15/30/60 slugs in favour for a single one; but the old urls are still detected properly as dynamic and will also work. So any slug can be used in fact. I also removed the description from the event.

**Review by CarinaWolli — DISMISSED** [review]
LGTM! Thank you 🙏

**Review by emrysal — DISMISSED** [review]
Lgtm!

**Comment by hussamkhatib** [comment]
Hey @emrysal , 
just curious, why can't we rely on [useRouter for updating the URL](https://github.com/calcom/cal.com/pull/9931/commits/e366122ea32508ad33363cad0e35df75cf32365b) ?
Wouldn't selecting a different time slot than the default show wrong URL params

**Comment by emrysal** [comment]
@hussamkhatib Not anymore (as I merged all 3 events)

The reason is because @calcom/atoms is intended to be used outside NextJS as well, so we're trying to keep NextJS and the Booker component uncoupled. useRouter is specific to NextJS so couldn't be used.

**Review by emrysal — APPROVED** [review]
